hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/listener/ClientSmartListenerService.java

```diff
@@ -88,364 +80,131 @@ public String call() {
+                    registrations.remove(key);
+                }
+                return successful;
             }
-        }
-        if (successful) {
-            registrations.remove(key);
-        }
-        return successful;
-    }
-
-    /**
-     * Should be called while in the registration executor
-     * @param registration The registration for which the deregister invocation will be sent
-     * @param subscriber The member added from which to deregister
-     * @return The future for the invocation.
-     */
-    private ClientInvocationFuture getClientInvocationFuture(ClientEventRegistration registration, Member subscriber) {
-        ClientInvocationFuture invocationFuture;
-        ListenerMessageCodec listenerMessageCodec = registration.getCodec();
-        String serverRegistrationId = registration.getServerRegistrationId();
-        ClientMessage request = listenerMessageCodec.encodeRemoveRequest(serverRegistrationId);
-        invocationFuture = new ClientInvocation(client, request, subscriber.getAddress()).invoke();
-
-        logger.finest("Invoked deregister listener invocation for " + registration + " to member " + subscriber);
-        return invocationFuture;
-    }
+        });
 
-    private Set<String> getMemberUuids() {
-        Set<String> memberUuids = new HashSet<String>(members.size());
-        for (Member m : members) {
-            memberUuids.add(m.getUuid());
+        try {
+            return future.get();
+        } catch (Exception e) {
+            throw ExceptionUtil.rethrow(e);
         }
-        return memberUuids;
     }
 
+
     @Override
     public void start() {
-        membershipListenerId = clusterService.addMembershipListener(this);
-        if (null != clusterService.getOwnerConnectionAddress()) {
-            lifecycleState = LifecycleEvent.LifecycleState.CLIENT_CONNECTED;
-        }
-        client.getLifecycleService().addLifecycleListener(new LifecycleListener() {
-            @Override
-            public void stateChanged(LifecycleEvent event) {
-                lifecycleState = event.getState();
-            }
-        });
-
-        connectionOpener = client.getClientExecutionService().scheduleWithRepetition(new Runnable() {
+        clientConnectionManager.addConnectionListener(this);
+        final ClientClusterService clientClusterService = client.getClientClusterService();
+        registrationExecutor.scheduleWithFixedDelay(new Runnable() {
             @Override
             public void run() {
-                registrationExecutor.submit(new Runnable() {
-                    @Override
-                    public void run() {
-                        ensureConnectionsToAllServers();
-                    }
-                });
+                Collection<Member> memberList = clientClusterService.getMemberList();
+                for (Member member : memberList) {
+                    clientConnectionManager.getOrTriggerConnect(member.getAddress(), false);
```

Comment:
There is a race of we call it with `true`. Only ClusterListenerSupport should call it with ownerConnection=true. In this case, there is no problem. If connection is established as owner to this address, it will return immediately without making a remote call, since it is already connected.